### PR TITLE
[ODS-4533] Migration Support for ODS / API 5.1.0

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1020-AddNewColumnsForV51.sql [hunk 1]
```diff
@@ -0,0 +1,775 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+--  Alter Table [edfi].[[AcademicWeek]]---
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1020-AddNewColumnsForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1030-AddNewDeleteTablesForV51.sql [hunk 2]
```diff
@@ -0,0 +1,2609 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE TABLE [tracked_deletes_edfi].[AbsenceEventCategoryDescriptor] (
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1030-AddNewDeleteTablesForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1040-CreateFunctionChangeGetMaxChangeVersionForV51.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+IF EXISTS (
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1040-CreateFunctionChangeGetMaxChangeVersionForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1100-AddNewTriggerForV51.sql [hunk 4]
```diff
@@ -0,0 +1,4918 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE TRIGGER [edfi].[edfi_AbsenceEventCategoryDescriptor_TR_DeleteTracking] ON [edfi].[AbsenceEventCategoryDescriptor] AFTER DELETE AS
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1100-AddNewTriggerForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1010-CreateSequenceForV51.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+IF NOT EXISTS (SELECT * FROM sys.sequences WHERE object_id = object_id('changes.ChangeVersionSequence'))
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1010-CreateSequenceForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1000-AddTrackedChangesSchemaForV51.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+IF NOT EXISTS (SELECT * FROM sys.schemas WHERE name = N'tracked_deletes_edfi')
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1000-AddTrackedChangesSchemaForV51.sql]: These is related to the change query feature. if you compare the database reference scripts in EdFi.Ods.Standard\Artifacts\MsSql\Structure\Ods\Changes for v5.0.0 and v.5.1.0 and don't see any differences we should remove this file.

**Review by tmeador-dl — CHANGES_REQUESTED** [review]
Looks good overall. You've added the change queries feature as a migration. Since change queries is off by default in v5.0.0 and then on by default in v5.1.0 it probably showed up in your schema compare as new tables. Since I don't see any differences in the reference scripts I think we can delete the files I've commented on below. Hopefully that will also fix the Grand Bend test failure we're seeing in the build.

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1040-CreateFunctionChangeGetMaxChangeVersionForV51.sql (hunk 3 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1040-CreateFunctionChangeGetMaxChangeVersionForV51.sql]: done. thanks

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1100-AddNewTriggerForV51.sql (hunk 4 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1100-AddNewTriggerForV51.sql]: done. thanks

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1010-CreateSequenceForV51.sql (hunk 5 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1010-CreateSequenceForV51.sql]: done. thanks

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1000-AddTrackedChangesSchemaForV51.sql (hunk 6 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1000-AddTrackedChangesSchemaForV51.sql]: done. thanks

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1030-AddNewDeleteTablesForV51.sql (hunk 2 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1030-AddNewDeleteTablesForV51.sql]: done. thanks

File: EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1020-AddNewColumnsForV51.sql (hunk 1 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/02Upgrade/v50_to_v51/01 Structure/1020-AddNewColumnsForV51.sql]: done. thanks

**Review by tmeador-dl — APPROVED** [review]

**Pull request merged**